pywbem_mock/_mockmofwbemconnection.py

```diff
@@ -28,30 +28,25 @@
 from __future__ import absolute_import, print_function
 
 from pywbem import CIMError, CIM_ERR_INVALID_PARAMETER, \
-    CIM_ERR_NOT_FOUND, CIM_ERR_ALREADY_EXISTS, CIM_ERR_FAILED, \
-    CIM_ERR_INVALID_SUPERCLASS, CIM_ERR_INVALID_NAMESPACE, CIMInstanceName
+    CIM_ERR_NOT_FOUND, CIM_ERR_FAILED, CIM_ERR_INVALID_SUPERCLASS
 from pywbem._nocasedict import NocaseDict
-from pywbem._mof_compiler import MOFWBEMConnection
+from pywbem._mof_compiler import BaseRepositoryConnection
 from pywbem._utils import _format
 from ._resolvermixin import ResolverMixin
 
 # Issue #2062 TODO/ks Remove this code and use the methods in _WBEMConnection
 # in their place
 
 
-class _MockMOFWBEMConnection(MOFWBEMConnection, ResolverMixin):
+class _MockMOFWBEMConnection(BaseRepositoryConnection, ResolverMixin):
```

Comment:
Why is this class needed at all?

Reply:
This class differs from the MOFWBEMConnection in that:

1. It requires the connection object, the other one treats conn as an option
2. It implements ModifyClass, MofWBEMConnection does not.
3. In only uses the local cache for classes when the CreateClass is trying to resolve dependent classes

In reality, I would assume we could bring them together but that is one more step.

**DISCUSSION**

Reply:
With the change of the superclass, they are going further apart rather than closer together.

Reply:
We discussed this today in a call and decided that getting proper user-provider support when the MOF compiler works against a fake connection is more important than loosing the rollback support. Withdrawing the comment. DONE.